qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -149,60 +227,99 @@ def settings(self) -> Dict[str, Any]:
             "callback": self.callback,
         }
 
-    def minimize(
+    def ask(self) -> AskObject:
+        """
+        For gradient descent this method simply returns an
+        :class:`qiskit.algorithms.optimizers.AskObject` containing the current point for the
+        gradient to be evaluated, a parameter epsilon in case the gradient has to be approximated
+        and a bool flag telling the prefered way of evaluating the gradient.
+        """
+        return AskObject(
+            x_jac=self._state.x,
+        )
+
+    def tell(self, ask_object: AskObject, tell_object: TellObject) -> None:
+        """
+        For gradient descent this method updates :attr:`.~GradientDescentState.x` by an ammount
+        proportional to the learning rate and the gradient at that point.
+        """
+        self._state.x = self._state.x - next(self._state.eta) * tell_object.eval_jac
+        self._state.stepsize = np.linalg.norm(tell_object.eval_jac)
+        self._state.nit += 1
+
+    def evaluate(self, ask_object: AskObject) -> TellObject:
+        """
+        Evaluates the gradient, either by evaluating an analitic gradient or by approximating it with a
+        finite difference scheme.
+        The value of the gradient is returned as a :class:`qiskit.algorithms.optimizers.TellObject`.
+        """
+        if self._state.jac is None:
+            eps = 0.01 if self.perturbation is None else self.perturbation
+            grad = Optimizer.gradient_num_diff(
+                x_center=ask_object.x_jac,
+                f=self._state.fun,
+                epsilon=eps,
+                max_evals_grouped=self._max_evals_grouped,
+            )
+            self._state.nfev += 1 + len(ask_object.x_jac)
+        else:
+            grad = self._state.jac(ask_object.x_jac)
+            self._state.njev += 1
+
+        return TellObject(eval_jac=grad)
+
+    def create_result(self) -> OptimizerResult:
+        """Creates a result of the optimization process using the values from
+        :class:`~.GradientDescentState`.
```

Comment:
Does this need to be a public method? The docstring seems off - the method returns an OptimizerResult - why does an end user care (assuming this is intended to be a public method) how it was internally created from a gradient descent state?

Reply:
I think the user should be able to create the result themselves in case they are doing a step by step optimization and for any reason they still want to have a result object.
I  will update the documentation.